frontend-react/src/components/FileHandlers/FileHandler.tsx

```diff
@@ -91,25 +106,54 @@ export function getClientHeader(
 
 function FileHandler() {
     const { state, dispatch } = useFileHandler();
-    const [fileContent, setFileContent] = useState("");
-
     const {
         fileInputResetValue,
         contentType,
         fileType,
         fileName,
         errors,
+        errorType,
         destinations,
         reportId,
         successTimestamp,
-        cancellable,
-        errorType,
         warnings,
         localError,
-        overallStatus,
         reportItems,
         selectedSchemaOption,
     } = state;
+    const [fileContent, setFileContent] = useState("");
+    const [fileHandlerStepIndex, setFileHandlerStepIndex] = useState(0);
+    // The File Validate tool now has 4 discrete steps,
+    // Schema Select, File Select, [optional]Show Errors, Success Page
+    // The stages can be seen here: https://figma.fun/fGCeo4
+    const fileHandlerStep = fileHandlerStepsArray[fileHandlerStepIndex];
+    const resetState = () => {
+        setFileContent("");
+        dispatch({ type: FileHandlerActionType.RESET });
+    };
+
+    const handlePrevFileHandlerStep = () => {
+        // We have to clear different parts of state depending
```

Comment:
I'd argue that these should be discrete actions that can be dispatched rather than handling this entirely in an `if-else`.